src/python/gudhi/point_cloud/timedelay.py

```diff
@@ -0,0 +1,56 @@
+import numpy as np
+
+class TimeDelayEmbedding:
+    """Point cloud transformation class.
+
+    Embeds time-series data in the R^d according to Takens' Embedding Theorem
+    and obtains the coordinates of each point.
+
+    Parameters
+    ----------
+    dim : int, optional (default=3)
+        `d` of R^d to be embedded.
+
+    delay : int, optional (default=1)
+        Time-Delay embedding.
+
+    skip : int, optional (default=1)
+        How often to skip embedded points.
+
+    """
+    def __init__(self, dim=3, delay=1, skip=1):
+        self._dim = dim
+        self._delay = delay
+        self._skip = skip
+
+    def __call__(self, *args, **kwargs):
+        return self.transform(*args, **kwargs)
+
+    def _transform(self, ts):
+        """Guts of transform method."""
+        return ts[
+            np.add.outer(
+                np.arange(0, len(ts)-self._delay*(self._dim-1), self._skip),
+                np.arange(0, self._dim*self._delay, self._delay))
+        ]
+
+    def transform(self, ts):
+        """Transform method.
+
+        Parameters
+        ----------
+        ts : list[float] or list[list[float]]
+            A single or multiple time-series data.
+
+        Returns
+        -------
+        point clouds : list[list[float, float, float]] or list[list[list[float, float, float]]]
+            Makes point cloud every a single time-series data.
+        """
+        ndts = np.array(ts)
+        if ndts.ndim == 1:
+            # for single.
+            return self._transform(ndts).tolist()
```

Comment:
Why convert everything to lists? Why not return the numpy array?

Reply:
When using it alone from the outside, I felt it was better to pass it in an   
orthodox LIST type (Python standard array).  
Until now, it was designed as an inner class so that it could not be used   
from the outside, so we passed data with ndarray.

Reply:
I believe numpy arrays (or at least buffer-compatible objects) are the standard when it comes to scientific computing. Scikit-learn uses them as input and output everywhere. Converting back and forth with lists is going to be costly.